cloud/amazon/ec2_ami.py

```diff
@@ -544,14 +644,49 @@ def main():
             # Update image's launch permissions
             update_image(module, ec2)
 
-        # Changed is always set to true when provisioning new AMI
-        if not module.params.get('instance_id'):
-            module.fail_json(msg='instance_id parameter is required for new image')
+        if module.params.get('action') not in ['create','register']:
+            module.fail_json(msg='action must be create or register')
+
+        if module.params.get('action') == "create":
+            # Changed is always set to true when provisioning new AMI
+            if not module.params.get('instance_id'):
+                module.fail_json(msg='instance_id is required for new image')
+
+        if module.params.get('action') == "register":
+            if not module.params.get('virtualization_type'):
+                module.fail_json(
+                    msg='virtualization_type is required for new image')
+
+            if module.params.get('virtualization_type') not in ['paravirtual', 'hvm']:
```

Comment:
Things like this should be set in the params above using the `choices` option to limit the selections.